app/schema_validation/__init__.py

```diff
@@ -65,39 +65,60 @@ def validate(
-            # Redact "personalisation"
-            if 'personalisation' in json_to_validate:
-                if isinstance(json_to_validate.get('personalisation'), dict):
-                    json_to_validate['personalisation'] = {
-                        key: '<redacted>' for key in json_to_validate['personalisation']
-                    }
-                else:
-                    json_to_validate['personalisation'] = '<redacted>'
-
-            # Redact ICN
-            if 'recipient_identifier' in json_to_validate:
-                if (
-                    isinstance(json_to_validate.get('recipient_identifier'), dict)  # Short circuit dictionary check
-                    and json_to_validate['recipient_identifier'].get('id_type') == 'ICN'
-                ):
-                    json_to_validate['recipient_identifier']['id_value'] = '<redacted>'
-                else:
-                    json_to_validate['recipient_identifier'] = '<redacted>'
+        # Redact "personalisation"
+        if 'personalisation' in json_to_validate:
+            if isinstance(json_to_validate.get('personalisation'), dict):
+                json_to_validate['personalisation'] = {key: '<redacted>' for key in json_to_validate['personalisation']}
+            else:
+                json_to_validate['personalisation'] = '<redacted>'
+
+        # Redact ICN
+        if 'recipient_identifier' in json_to_validate:
+            if (
+                isinstance(json_to_validate.get('recipient_identifier'), dict)  # Short circuit dictionary check
+                and json_to_validate['recipient_identifier'].get('id_type') == 'ICN'
+            ):
+                json_to_validate['recipient_identifier']['id_value'] = '<redacted>'
+            else:
+                json_to_validate['recipient_identifier'] = '<redacted>'
         current_app.logger.info('Validation failed for: %s', json_to_validate)
         raise ValidationError(build_error_message(errors))
 
-    # TODO - This assumes that json_to_validate is a dictionary.  It could raise AttributeError.
-    if json_to_validate.get('personalisation'):
-        json_to_validate['personalisation'], errors = decode_personalisation_files(
-            json_to_validate.get('personalisation', {})
-        )
-        if len(errors) > 0:
-            error_message = json.dumps({'status_code': 400, 'errors': errors})
-            raise ValidationError(error_message)
+    try:
+        if json_to_validate.get('personalisation'):
+            json_to_validate['personalisation'], errors = decode_personalisation_files(
+                json_to_validate.get('personalisation', {})
+            )
+            if len(errors) > 0:
+                error_message = json.dumps({'status_code': 400, 'errors': errors})
+                raise ValidationError(error_message)
+    except AttributeError:
+        current_app.logger.info('Validation failed for: %s', json_to_validate)
+        raise ValidationError('Payload is not json.')
```

Comment:
You don't need try/except here.  You already tested above that json_to_validate is a dictionary.